components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +325,79 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  private showInputs(fieldName: string): void {
+
+    const formValues = this.expressionForm.value;
+    const matchAllWildCard = '*';
+
+    this.setFormLabels(formValues.type);
+
+    switch (fieldName) {
+      case 'type':
+        this.resetFormControls();
+        if (formValues.type === 'user' || formValues.type === 'team') {
+          this.addIdentityControl();
+        } else if (formValues.type === 'token') {
+          this.addNameControl();
+        }
+        break;
+      case 'identity':
+        if (formValues.identity !== matchAllWildCard) {
+          this.addNameControl();
+        } else {
+          this.expressionForm.removeControl('name');
+        }
+        break;
+      default:
```

Comment:
Right now the reader may wonder whether or not `name` was left out deliberately or not. Suggest add `case 'name'` just above the default so it is clear that for `name` we need do nothing.